azurelinuxagent/common/osutil/default.py

```diff
@@ -630,6 +630,11 @@ def umount_dvd(self, chk_err=True, mount_point=None):
 
     def eject_dvd(self, chk_err=True):
         dvd = self.get_dvd_device()
+        dev = dvd.rsplit('/',1)[1]
+        pattern = r'(vd[b-z])'
+        # We should not eject if the disk is not a cdrom
+        if re.search(pattern, dev):
```

Comment:
Why do we not eject this? Is this a device that can't be ejected?

Reply:
Yes. You are right. this is the disk that shows up as a regular attached disk. This disk we pass to the vmm as a virtio disk, not a cdrom. That can't be ejected. Most of the Cloud vmms in the open-source world don't have any CDrom support. SO we are planning to pass virtio disk for he configs instead of cdrom.

Reply:
Can there be a scenario where a vd[b-z] device contains the OVF but it actually is ejectable? I think it would be better keep the logic here and ignore errors during ejection (or log a message and move on)

Reply:
Another question: I assume the platform in these cases will remove the underlying device at ready signal, or after some amount of time (similar to how it is today with the DVD ISO)?

Reply:
> Another question: I assume the platform in these cases will remove the underlying device at ready signal, or after some amount of time (similar to how it is today with the DVD ISO)?

YOu are right.

Reply:
> Can there be a scenario where a vd[b-z] device contains the OVF but it actually is ejectable? I think it would be better keep the logic here and ignore errors during ejection (or log a message and move on)

vd[b-z]  will not be ejectable.

Reply:
@anhvoms We have notified our partner team to address the security issue with plain text credentials. Can we move forward with this PR?

Reply:
Yes this is good to move forward